src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -94,6 +96,60 @@ async def ensure_executor_health(self):
 
 
 class APIBasedExecutorProxy(AbstractExecutorProxy):
+    def __init__(
+        self,
+        *,
+        working_dir: Path = None,
+    ):
+        """Initialize the executor proxy with the working directory.
+
+        :param working_dir: The working directory of the executor, usually the flow directory,
+                where we can find metadata under .promptflow. Will use current working directory if not provided.
+        :type working_dir: Path
+        """
+        self._working_dir = working_dir or Path.cwd()
+
+    @property
+    def working_dir(self) -> Path:
+        """
+        The working directory of the executor, usually the flow directory,
+        where we can find metadata under .promptflow.
+        """
+        return self._working_dir
+
+    def _get_flow_meta(self) -> dict:
+        flow_meta_json_path = self.working_dir / PROMPT_FLOW_DIR_NAME / FLOW_META_JSON
```

Comment:
When flow changed, when will the meta file be regenerated?

Reply:
proxy just expose an interface and the generation will be controlled by SDK